provider/spark.go

```diff
@@ -626,8 +630,172 @@ func (e EMRExecutor) InitializeExecutor(store FileStore) error {
 	return store.Write(sparkScriptPath, buff)
 }
 
-func NewSparkExecutor(execType SparkExecutorType, config SparkExecutorConfig, logger *zap.SugaredLogger) (SparkExecutor, error) {
+type SparkGenericConfig struct {
+	Master        string
+	DeployMode    string
+	PythonVersion string
+}
+
+func (sc *SparkGenericConfig) Deserialize(config SerializedConfig) error {
+	err := json.Unmarshal(config, sc)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sc *SparkGenericConfig) Serialize() ([]byte, error) {
+	conf, err := json.Marshal(sc)
+	if err != nil {
+		return nil, err
+	}
+	return conf, nil
+}
+
+func (sc *SparkGenericConfig) IsExecutorConfig() bool {
+	return true
+}
+
+type SparkGenericExecutor struct {
+	master        string
+	deployMode    string
+	pythonVersion string
+	logger        *zap.SugaredLogger
+}
+
+func (s *SparkGenericExecutor) InitializeExecutor(store FileStore) error {
+	s.logger.Info("Uploading PySpark script to filestore")
+	sparkScriptPath := helpers.GetEnv("SPARK_SCRIPT_PATH", "/scripts/offline_store_spark_runner.py")
+	sparkScriptPathWithPrefix := store.PathWithPrefix(sparkScriptPath, false)
+
+	err := readAndUploadFile(sparkScriptPath, sparkScriptPathWithPrefix, store)
+	scriptExists, _ := store.Exists(sparkScriptPathWithPrefix)
+	if err != nil && !scriptExists {
+		return fmt.Errorf("could not upload spark script: Path: %s, Error: %v", sparkScriptPathWithPrefix, err)
+	}
+	return nil
+}
+
+func (s *SparkGenericExecutor) RunSparkJob(args *[]string, store FileStore) error {
+	bashCommand := "bash"
+	sparkArgs := *args
+	sparkArgsString := strings.Join(sparkArgs, " ")
+	bashCommandArgs := []string{"-c", fmt.Sprintf("pyenv global %s && pyenv exec %s", s.pythonVersion, sparkArgsString)}
 
+	s.logger.Infow("Executing the", bashCommand, "command")
+	cmd := exec.Command(bashCommand, bashCommandArgs...)
+	cmd.Env = append(os.Environ(), "FEATUREFORM_LOCAL_MODE=true")
+
+	err := cmd.Start()
+	if err != nil {
+		return fmt.Errorf("could not run spark job: %v", err)
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		return fmt.Errorf("spark job failed: %v", err)
+	}
+
+	return nil
+}
+
+func (s *SparkGenericExecutor) PythonFileURI(store FileStore) string {
+	return ""
+}
+
+func (s *SparkGenericExecutor) SparkSubmitArgs(destPath string, cleanQuery string, sourceList []string, jobType JobType, store FileStore) []string {
+	sparkScriptPath := helpers.GetEnv("SPARK_SCRIPT_PATH", "/scripts/offline_store_spark_runner.py")
+
+	argList := []string{
+		"spark-submit",
+		"--master",
+		s.master,
+		"--deploy-mode",
+		s.deployMode,
+		"--packages",
+		"\"org.apache.hadoop:hadoop-azure:3.2.0\"",
```

Comment:
?

Reply:
needed for reading data from azure blob store.

Reply:
But why add it to the main arg list why not add it to the ones you do if its azure